src/NexusMods.App.UI/Pages/Sorting/LoadOrder/TreeDataGrid/LoadOrderDataProvider.cs

```diff
@@ -1,26 +1,56 @@
+using System.ComponentModel;
 using System.Reactive.Linq;
 using DynamicData;
 using DynamicData.Binding;
 using Humanizer;
 using NexusMods.Abstractions.Games;
 using NexusMods.App.UI.Controls;
+using R3;
 using ReactiveUI;
 using static NexusMods.App.UI.Pages.Sorting.LoadOrderComponents;
+// ReSharper disable InvokeAsExtensionMethod
 
 namespace NexusMods.App.UI.Pages.Sorting;
 
 public class LoadOrderDataProvider : ILoadOrderDataProvider
 {
     
-    public IObservable<IChangeSet<CompositeItemModel<Guid>, Guid>> ObserveLoadOrder(ILoadoutSortableItemProvider sortableItemProvider)
+    public IObservable<IChangeSet<CompositeItemModel<Guid>, Guid>> ObserveLoadOrder(
+        ILoadoutSortableItemProvider sortableItemProvider,
+        Observable<ListSortDirection> sortDirectionObservable)
     {
+        var lastIndexObservable = sortableItemProvider.SortableItems
+            .ToObservableChangeSet(item => item.ItemId)
```

Comment:
You're using `ToObservableChangeSet` in a couple of places already. It's a fairly expensive method, I'd suggest either exposing a source cache or a connectable observable on the sortable item provider.

Reply:
That is a good idea, will do in a separate PR.